tests/specs/lemmas.k

```diff
@@ -65,6 +72,29 @@ module LEMMAS [symbolic]
     rule WS         ++ .ByteArray => WS  [simplification]
     rule .ByteArray ++ WS         => WS  [simplification]
 
+    // #range
+
+    rule #range(_, _, K) => 
+         .ByteArray 
+      requires notBool K >Int 0 [simplification]
+
+    rule #range(M [ N := BUF:ByteArray ], L, K) => 
+         #range(M, L, minInt(K, N -Int L)) ++ #range(M [ N := BUF ], N, K -Int minInt(K, N -Int L)) 
+      requires K  >Int 0 andBool L <Int N [simplification]
```

Comment:
```suggestion
    rule #range(M [ N := BUF:ByteArray ], L, K) => 
         #let W = minInt(K, N -Int L) #in #range(M, L, W) ++ #range(M [ N := BUF ], N, K -Int W)
      requires K  >Int 0 andBool L <Int N [simplification]
```

Reply:
@PetarMax you may be interested in this one as well, `#let` is not documented in K tutorial or User manual.

Reply:
Thanks, that's a very useful feature I wasn't aware of.

Reply:
I changed this to use the let binding.